### PR TITLE
fix: revert sandbox lite image tag prefix to sha

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Currently we are not publishing sandbox lite image with `dind` tag prefix, reverted sandbox lite image prefix tag to `sha`

## Type
- [x] Bug Fix
- [ ] Feature
- [ ] Plugin

## Are all requirements met?

- [ ] Code completed
- [ ] Smoke tested
- [ ] Unit tests added
- [ ] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
_How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: cmd/demo/start.go [hunk 1]
```diff
@@ -40,19 +40,19 @@ Starts the demo cluster without any source code:
...
+ flytectl demo start --source=$HOME/flyteorg/flytesnacks
 
 Specify a Flyte demo compliant image with the registry. This is useful in case you want to use an image from your registry.
 ::
 
   flytectl demo start --image docker.io/my-override:latest
 
 Note: If image flag is passed then Flytectl will ignore version and pre flags.
-	
+
```
Comment (pmahindrakar-oss) [inline cmd/demo/start.go:55]: Can we add example usage with version  & prerelease with some actual values

File: cmd/sandbox/start.go [hunk 2]
```diff
@@ -191,7 +191,7 @@ func startSandbox(ctx context.Context, cli docker.Docker, reader io.Reader) (*bu
 	}
 	sandboxImage := sandboxConfig.DefaultConfig.Image
 	if len(sandboxImage) == 0 {
-		image, version, err := githubutil.GetFullyQualifiedImageName(sandboxConfig.DefaultConfig.Version, sandboxImageName, sandboxConfig.DefaultConfig.Prerelease)
+		image, version, err := githubutil.GetFullyQualifiedImageName("dind", sandboxConfig.DefaultConfig.Version, sandboxImageName, sandboxConfig.DefaultConfig.Prerelease)
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:194]: Instead of passing it can we add this to the config .
ie have the 
```
var (
	DefaultSingleBinaryConfig = &Config{
   prefix : "sha"
}
DefaultNonSingleBinaryConfig = &Config{
   prefix : "dind"
}
)

```
type Config struct {
...
	Prefix  string `json:"prefix" pflag:-", "` // dont generate pflag for this
}
```

Comment (evalsocket) [reply cmd/sandbox/start.go:194]: that is not needed because it's the past debt we have on our sandbox image, We never user non dind image anywhere, we want to move on sha prefix for sandbox. 

we have `--image` flag for specifying the image full name

File: cmd/demo/start.go [hunk 3]
```diff
@@ -40,31 +40,43 @@ Starts the demo cluster without any source code:
...
 Specify a Flyte demo image pull policy. Possible pull policy values are Always, IfNotPresent, or Never:
 ::
 
  flytectl demo start  --image docker.io/my-override:latest --imagePullPolicy Always
 
+Runs a specific version of Flyte. Flytectl sandbox only supports Flyte version available in the Github release, https://github.com/flyteorg/flyte/tags.
+::
+
+ flytectl sandbox start  --version=v0.14.0
```
Comment (eapolinario) [inline cmd/demo/start.go:64]: `flytectl demo start`

File: cmd/demo/start.go [hunk 4]
```diff
@@ -40,31 +40,43 @@ Starts the demo cluster without any source code:
...
+
+.. note::
+	  Flytectl Sandbox is only supported for Flyte versions > v1.0.0
+
+Runs the latest pre release of  Flyte.
+::
+
+ flytectl sandbox start  --pre
```
Comment (eapolinario) [inline cmd/demo/start.go:72]: ditto.

**Review by eapolinario — COMMENTED** [review]
Two minor comments. 

I tested it locally too for good measure.

**Review by kumare3 — APPROVED** [review]
